astropy/io/fits/convenience.py

```diff
@@ -463,6 +462,8 @@ def table_to_hdu(table):
 
     # Not all tables with mixin columns are supported
     if table.has_mixin_columns:
+        from ...table.column import BaseColumn
```

Comment:
Done these changes in the latest commit. And yes it will become second nature soon 😄